src/tests/ftest/telemetry/pool_svc_metrics.py

```diff
@@ -0,0 +1,133 @@
+                            continue
+                        if 'pool' not in m['labels'] or \
+                                m['labels']['pool'].casefold() != str(pool_uuid).casefold():
+                            continue
+                        if 'rank' not in m['labels']:
+                            continue
+                        rank = m['labels']['rank']
+                        if rank not in pm:
+                            pm[rank] = {}
+                        pm[rank][k] = m['value']
+            return pm
+
+        metrics_list = ",".join(self.telemetry.ENGINE_POOL_SVC_METRICS)
+        host_metrics = self.telemetry.get_metrics(metrics_list)
+        self.log.debug("host metrics: %s", host_metrics)
+        pr_metrics = _pool_rank_metrics(host_metrics, pool_uuid)
+        self.log.debug("pool rank metrics: %s", pr_metrics)
+        leader_metrics = {}
+        for metrics in pr_metrics.values():
+            if _map_version(metrics) > _map_version(leader_metrics):
+                leader_metrics = metrics
+
+        self.log.debug("Pool service leader metrics: %s", leader_metrics)
+        return leader_metrics
+
+    def test_pool_service_metrics(self):
+        """Test that pool service telemetry is updated as expected.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=telemetry
+        :avocado: tags=PoolServiceMetrics,test_pool_service_metrics
+        """
+        self.log_step("Create pool for testing.")
+        pool = self.get_pool(connect=True)
+
+        self.log_step("Collect pool service metrics prior to making changes.")
+        initial_metrics = self.collect_svc_telemetry(pool.uuid)
+        self.assertTrue(MAP_VERSION_METRIC in initial_metrics,
+                        f"initial metrics don't contain {MAP_VERSION_METRIC} (no leader?)")
+        self.assertTrue(initial_metrics[MAP_VERSION_METRIC] == 1,
+                        "initial pool service map version is not 1")
+        self.assertTrue(initial_metrics[DEGRADED_RANKS_METRIC] == 0,
+                        "initial pool service degraded rank count is not 0")
+
+        restart_rank = initial_metrics[SVC_LEADER_METRIC]
+        self.log_step(f"Stop pool service leader rank: {restart_rank}")
+        self.server_managers[0].stop_ranks(ranks=[restart_rank], daos_log=self.d_log)
+
+        self.log_step("Verify the pool service leader rank has stopped successfully.")
+        failed_ranks = self.server_managers[0].check_rank_state(
+            ranks=[restart_rank], valid_states=["stopped", "excluded"], max_checks=15)
+        if failed_ranks:
+            self.fail(f"Rank {restart_rank} didn't stop!")
+
+        self.log_step("Wait for rank exclusion to show up in the telemetry.")
+        metrics = self.collect_svc_telemetry(pool.uuid)
+        while True:
+            if MAP_VERSION_METRIC in metrics and metrics[MAP_VERSION_METRIC] > 1:
+                break
+
+            time.sleep(5)
```

Comment:
Please add a log message that we're sleeping here so the logs don't look like something is hanging (it's been an issue before)

Reply:
Will do.